File: mobile/src/main/java/com/keepnotes/storage/IDataManager.java
package com.keepnotes.storage;

import com.keepnotes.models.Note;

import java.util.ArrayList;

@SuppressWarnings("unused")
public interface IDataManager {
    ArrayList<Note> getNotes();

    void setNotes(ArrayList<Note> notes);

    void removeNote(Note note);

    void saveNote(Note note);

    void changeNoteStatus(Note note);
}

File: mobile/src/main/java/com/keepnotes/storage/IDataManager.kt
package com.keepnotes.storage

import com.keepnotes.models.Note
import java.util.*

interface IDataManager {
    fun getNotes(): ArrayList<Note>

    fun setNotes(notes: ArrayList<Note>)

    fun removeNote(note: Note)

    fun saveNote(note: Note)

    fun changeNoteStatus(note: Note)
}
